src/core/internal/traits.d

```diff
@@ -88,7 +88,12 @@ private template substInoutForm(T)
 /// used to declare an extern(D) function that is defined in a different module
 template externDFunc(string fqn, T:FT*, FT) if(is(FT == function))
 {
-    static if (is(FT RT == return) && is(FT Args == function))
+    static if (__traits(compiles, __traits(externDmangle, "foo", void function())))
+    {
+        alias _wrap(alias T) = T; // fool the parser
+        alias externDFunc = _wrap!(__traits(externDmangle, fqn, FT));
```

Comment:
Unfortunately the qualified name in mangleFunc is a runtime parameter. No luck using a trait there...